edb/server/protocol/auth_ext/oauth.py

```diff
@@ -30,44 +30,39 @@ class Client:
     provider: base.BaseProvider
 
     def __init__(
-        self,
-        db: Any,
-        provider_name: str,
-        base_url: str | None = None
+        self, db: Any, provider_name: str, base_url: str | None = None
     ):
         self.db = db
 
         http_factory = lambda *args, **kwargs: http_client.HttpClient(
             *args, edgedb_test_url=base_url, **kwargs
         )
 
+        provider_args = self._get_provider_config(provider_name)
+        provider_kwargs = {
+            "http_factory": http_factory,
+            "additional_scope": self._get_additional_scope(provider_name),
```

Comment:
Can we fold `_get_additional_scope` into `_get_provider_config`? It's not the fastest APIs I think

Reply:
It's a little tricky because we're returning a tuple and `additional_scope` is a keyword-only argument, but I think I can still do that. Will fix.

Reply:
@1st1 ⚡ [`948aaa4` (#6234)](https://github.com/edgedb/edgedb/pull/6234/commits/948aaa4b9fab456b642c5481dc1b8b6707389b65)

How does this look?

Reply:
Better! You can also just return `cfg` from that method, that would probably be more future proof, but up to you. This PR is good to go as far as I'm concerned

Reply:
That _does_ seem better. Should I make a dataclass for `ProviderConfig` to get proper typing when accessing the properties of the returned config object?